### PR TITLE
Set up Application Insights

### PR DESCRIPTION
**Pull request by erbridge**
We should see request logging show up once it's deployed. This doesn't do any more than that on the AI front.

This also fixes the deployment of the tmp resource groups. You aren't allowed to have more than one app service with a given host name, so we disable deployment of the hostname parts when deploying a tmp app. The dependency graph was also missing some explicit dependencies that blocked a fresh deployment.

### CODE REVIEW

**Comment by erbridge** [comment]
[EPIC : Set up Application Insights on CIP](https://trello.com/c/oiJ1sy1k/251-epic-set-up-application-insights-on-cip)

File: config/initializers/application_insights.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+unless Rails.env.test?
```
Comment (tekin) [inline config/initializers/application_insights.rb]: Would this be better as `if Rails.env.production?` given that the Azure environments will run in that environment? Or is there is a specific reason we'd want to try and load application insights in someone's local development environment?

Comment (erbridge) [reply config/initializers/application_insights.rb]: @pezholio removed the check entirely. I assume that was discussed.

Comment (tekin) [reply config/initializers/application_insights.rb]: Yep, I'm happy from my end. might want to check in with @pezholio who's working with Craig today.

**Pull request merged**